### PR TITLE
(#141) Redirect NoTrials from a non-existent page fix

### PR DESCRIPTION
**Pull request by alerosad**
- Disease redirects to NoTrials now
- Intervention redirects to NoTrials now
- Manual redirects to NoTrials now
- having issue with integration test

### CODE REVIEW

**Comment by alerosad** [comment]
Valid page number:

https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-144/breast-cancer?cfg=0

Invalid page number:

https://react-app-dev.cancer.gov/clinical-trials-listing-app/pr-144/breast-cancer?cfg=0&pn=900

**Comment by kate-mashkina** [comment]
@alinai

File: src/views/Disease/Disease.jsx [hunk 1]
```diff
@@ -121,7 +121,42 @@ const Disease = ({ routeParamMap, routePath, data }) => {
 	const fetchState = useCtsApi(requestQuery);
 
 	useEffect(() => {
-		if (!fetchState.loading && fetchState.payload) {
+		//if you try to access a nonexistent page, eg: try to access page 41 of 1-40 pages
+		console.log('huh');
```
Comment (kate-mashkina) [inline src/views/Disease/Disease.jsx:125]: remove this logging

File: src/views/Disease/Disease.jsx [hunk 2]
```diff
@@ -121,7 +121,42 @@ const Disease = ({ routeParamMap, routePath, data }) => {
...
+
+		if (
+			!fetchState.loading &&
+			fetchState.error != null &&
+			fetchState.error.message === 'Trial count mismatch from the API'
+		) {
+			console.log('major yuh');
```
Comment (kate-mashkina) [inline src/views/Disease/Disease.jsx:132]: and this

File: src/views/Disease/Disease.jsx [hunk 3]
```diff
@@ -121,7 +121,42 @@ const Disease = ({ routeParamMap, routePath, data }) => {
...
+
+			console.log('got here i guess');
```
Comment (kate-mashkina) [inline src/views/Disease/Disease.jsx:134]: loving your logs, but they got to go :)

File: src/views/Disease/Disease.jsx [hunk 4]
```diff
@@ -121,7 +121,42 @@ const Disease = ({ routeParamMap, routePath, data }) => {
...
+				{
+					replace: true,
+					state: {
+						redirectStatus: redirectStatusCode,
+						prerenderLocation: prerenderLocation,
+					},
+				}
+			);
+		} else if (!fetchState.loading && fetchState.payload) {
+			console.log('in second statement');
```
Comment (kate-mashkina) [inline src/views/Disease/Disease.jsx:159]: here as well

**Review by kate-mashkina — CHANGES_REQUESTED** [review]

**Review by KateMashkinaNIH — APPROVED** [review]

**Comment by KateMashkinaNIH** [comment]
tested on ode450 for disease, intervention and 1 manual page

**Pull request merged**